pkg/cortex/modules.go

```diff
@@ -650,24 +658,26 @@ func (t *Cortex) setupModuleManager() error {
 
 	// Add dependencies
 	deps := map[string][]string{
-		API:            {Server},
-		Ring:           {API, RuntimeConfig, MemberlistKV},
-		Overrides:      {RuntimeConfig},
-		Distributor:    {Ring, API, Overrides},
-		Store:          {Overrides, DeleteRequestsStore},
-		Ingester:       {Overrides, Store, API, RuntimeConfig, MemberlistKV},
-		Flusher:        {Store, API},
-		Querier:        {Overrides, Distributor, Store, Ring, API, StoreQueryable, MemberlistKV},
-		StoreQueryable: {Overrides, Store},
-		QueryFrontend:  {API, Overrides, DeleteRequestsStore},
-		TableManager:   {API},
-		Ruler:          {Overrides, Distributor, Store, StoreQueryable, RulerStorage},
-		Configs:        {API},
-		AlertManager:   {API},
-		Compactor:      {API, MemberlistKV},
-		StoreGateway:   {API, Overrides, MemberlistKV},
-		Purger:         {Store, DeleteRequestsStore, API},
-		All:            {QueryFrontend, Querier, Ingester, Distributor, TableManager, Purger, StoreGateway, Ruler},
+		API:                {Server},
+		Ring:               {API, RuntimeConfig, MemberlistKV},
+		Overrides:          {RuntimeConfig},
+		Distributor:        {DistributorService, DistributorAPI},
```

Comment:
Do we need both `Distributor` and `DistributorAPI`?

Reply:
I think the `DistributorAPI` can be removed and the route registration can be moved to the `Distributor` module